runtime/near-vm-runner/src/preload.rs

```diff
@@ -41,27 +54,76 @@ pub struct ContractCallPrepareResult {
 
 pub struct ContractCaller {
     pool: ThreadPool,
-    prepared: Vec<CallInner>,
+    vm_kind: VMKind,
+    vm_config: VMConfig,
+    vm_data_private: VMDataPrivate,
+    vm_data_shared: VMDataShared,
+    preloaded: Vec<CallInner>,
 }
 
 impl ContractCaller {
-    pub fn new(num_threads: usize) -> ContractCaller {
-        ContractCaller { pool: ThreadPool::new(num_threads), prepared: Vec::new() }
+    pub fn new(num_threads: usize, vm_kind: VMKind, vm_config: VMConfig) -> ContractCaller {
+        let (shared, private) = match vm_kind {
+            VMKind::Wasmer0 => (
+                VMDataShared::Wasmer0,
+                VMDataPrivate::Wasmer0(if SHARE_MEMORY_INSTANCE {
+                    Some(
+                        WasmerMemory::new(
+                            vm_config.limit_config.initial_memory_pages,
+                            vm_config.limit_config.max_memory_pages,
+                        )
+                        .unwrap(),
```

Comment:
(unrelated to PR per se) urgh, we should pull these unwraps up, into `WasmerMemory::new`. They indeed can not fail as far as I can see, but we should assert that once.

Reply:
Interesting question, see benefits in either approach.